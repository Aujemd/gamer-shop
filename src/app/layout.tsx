import type { Metadata } from "next";

import { archivo, area } from "@/app/fonts";
import { Layout } from "@/app/layout/index";

import "../css/global.css";

export const metadata: Metadata = {
    title: "Apply Digital Test",
    description: "Frontend development test for Apply Digital",
};

/**
 * Root layout for the application.
 * This layout wraps all pages and provides a consistent structure.
 * @param props - The component props.
 * @param props.children - The child components to be rendered within this layout.
 * @returns The root layout component.
 */
export default function RootLayout({
    children,
}: Readonly<{
    children: React.ReactNode;
}>) {
    return (
        <html lang="en">
            <body className={`${archivo.variable} ${area.variable}`}>
                <Layout>
                    {children}
                </Layout>
            </body>
        </html>
    );
}
