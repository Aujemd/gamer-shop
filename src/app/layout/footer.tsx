import type { HTMLAttributes } from "react";

import Link from "next/link";

import { LogoIcon } from "@/icons";

type FooterProps = Readonly<Omit<HTMLAttributes<HTMLElement>, "children">>;

/**
 * Footer component for the application.
 * @param props - The component props.
 * @returns The footer component.
 */
export default function Footer(props: FooterProps) {
    const { className = "", ...rest } = props;
    return (
        <footer {...rest} className={`bg-neutral-700 py-16 flex justify-center items-center ${className}`}>
            <Link
                href="/"
                data-testid="footer-logo-link-test-id"
            >
                <LogoIcon
                    data-testid="footer-logo-icon-test-id"
                />
            </Link>
        </footer>
    );
}
