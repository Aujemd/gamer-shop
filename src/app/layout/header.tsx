import type { HTMLAttributes } from "react";

import Link from "next/link";

import { CartIcon } from "@/icons";

type HeaderProps = Readonly<Omit<HTMLAttributes<HTMLElement>, "children">>;

/**
 * Header component for the application.
 * @param props - The component props.
 * @returns The header component.
 */
export default function Header(props: HeaderProps) {
    const { className = "", ...rest } = props;
    return (
        <header {...rest} className={`bg-surface-secondary  ${className}`}>
            <div className="px-6 py-5 flex items-center justify-between text-cta-fill-primary lg:max-w-desktop mx-auto">
                <h1 className="font-area font-bold text-2xl leading-6 text-center tracking-sm">
                    <Link
                        href="/"
                        data-testid="header-logo-link-test-id"
                    >
                        GamerShop
                    </Link>
                </h1>
                <Link
                    href="/cart"
                    data-testid="header-cart-link-test-id"
                >
                    <CartIcon
                        data-testid="header-cart-icon-test-id"
                    />
                </Link>

            </div>

        </header>
    );
}
