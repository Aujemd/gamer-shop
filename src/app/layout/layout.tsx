import type { PropsWithChildren } from "react";

import { DataFetchingProvider, OrFooter, OrHeader } from "@/shared";

type LayoutProps = Readonly<PropsWithChildren>;

/**
 * Layout component for the application.
 * @param props - The component props.
 * @returns The layout component.
 */
export default function Layout(props: LayoutProps) {
    const { children } = props;
    return (

        <DataFetchingProvider>
            <OrHeader data-testid="layout-or-header-test-id" />
            <main className="min-h-screen">
                {children}
            </main>
            <OrFooter data-testid="layout-or-footer-test-id" />
        </DataFetchingProvider>

    );
}
