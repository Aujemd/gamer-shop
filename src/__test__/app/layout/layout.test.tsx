import { render } from "@testing-library/react";
import React from "react";

import { Layout } from "@/app/layout/index";

describe("layout Component", () => {
    it("renders children correctly", () => {
        const contentMock = "Hello Layout";
        const { asFragment, getByTestId } = render(
            <Layout>
                <div data-testid="child">{contentMock}</div>
            </Layout>,
        );
        expect(getByTestId("child")).toBeInTheDocument();
        expect(getByTestId("child")).toHaveTextContent(contentMock);
        expect(asFragment()).toMatchSnapshot();
    });
});
