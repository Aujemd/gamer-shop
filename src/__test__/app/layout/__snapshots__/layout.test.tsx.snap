// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout Component renders children correctly 1`] = `
<DocumentFragment>
  <main>
    <div
      data-testid="child"
    >
      Hello Layout
    </div>
  </main>
</DocumentFragment>
`;
